[Automation] native-image run fails for backport-util-concurrent:backport-util-concurrent:3.1

## native-image run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.1" ./gradlew clean test -Pcoordinates="backport-util-concurrent:backport-util-concurrent:3.1"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25727047580/job/75542580965
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
-d96f-ea51-4008dd46de1d \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature,com.oracle.svm.polyglot.scala.ScalaFeature'
]
========================================================================================================================
GraalVM Native Image: Generating 'backport-util-concurrent.backport-util-concurrent_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                    (6.2s @ 0.16GB)
 Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.0.3+9.1
 Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 C compiler: gcc (linux, x86_64, 13.3.0)
 Garbage collector: Serial GC (max heap size: 80% of RAM)
 2 user-specific feature(s):
 - com.oracle.svm.thirdparty.gson.GsonFeature
 - org.graalvm.junit.platform.JUnitPlatformFeature
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 12.67GB of memory (75.6% of system memory, $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                   (35.7s @ 0.86GB)
    8,262 types,   9,365 fields, and  37,069 methods found reachable
    2,828 types,     417 fields, and   3,005 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
        0 downcalls and 0 upcalls registered for foreign access
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                               (4.4s @ 0.97GB)
[4/8] Parsing methods...      [**]                                                                       (4.1s @ 0.60GB)
[5/8] Inlining methods...     [***]                                                                      (2.3s @ 0.66GB)
[6/8] Compiling methods...    [****]                                                                    (19.3s @ 0.81GB)
[7/8] Laying out methods...   [**]                                                                       (3.4s @ 0.76GB)
[8/8] Creating image...       [**]                                                                       (3.3s @ 0.90GB)
   9.86MB (36.34%) for code area:    23,502 compilation units
  13.50MB (49.74%) for image heap:  211,230 objects and 153 resources
   3.78MB (13.92%) for other data
  27.14MB in total image size, 25.04MB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   4.65MB java.base                                            2.86MB byte[] for code metadata
   2.19MB svm.jar (Native Image)                               2.56MB byte[] for java.lang.String
 588.81kB assertj-core-3.22.0.jar                              1.35MB java.lang.String
 481.60kB java.xml                                             1.03MB java.lang.Class
 427.77kB backport-util-concurrent-3.1.jar                   859.25kB com.oracle.svm.core.hub.DynamicHubCompanion
 198.42kB junit-jupiter-engine-5.9.2.jar                     707.49kB byte[] for general heap data
 123.28kB junit-platform-launcher-1.9.2.jar                  451.90kB java.util.concurrent.ConcurrentHashMap$Node
 104.28kB java.logging                                       434.26kB heap alignment
  99.84kB junit-platform-engine-1.9.2.jar                    301.02kB java.util.HashMap$Node
  82.60kB junit-4.13.2.jar                                   294.06kB byte[] for reflection metadata
 554.40kB for 24 more packages                                 2.65MB for 2010 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 19 component(s) is embedded in binary (1.00kB). 27 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                        4.9s (6.0% of total time) in 718 GCs | Peak RSS: 1.68GB | CPU load: 3.61
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/native/nativeTestCompile/backport-util-concurrent.backport-util-concurrent_tests-tests (executable)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library)
 /tmp/native-image-build-output.json (build_info)
========================================================================================================================
Finished generating 'backport-util-concurrent.backport-util-concurrent_tests-tests' in 1m 21s.
[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

backport_util_concurrent.backport_util_concurrent.AbstractExecutorServiceTest > bulkExecutionUsesAbstractExecutorServiceTaskManagement() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.AbstractExecutorServiceTest > shutdownRejectsNewTasks() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.AbstractExecutorServiceTest > submitWrapsRunnableAndCallableTasksInFutures() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ArrayBlockingQueueTest > typedArrayConversionAllocatesArrayWithComponentTypeAndQueueOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ArrayDequeTest > serializationRoundTripPreservesOrderAndDequeBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ArrayDequeTest > typedArrayConversionAllocatesArrayWithComponentTypeAndDequeOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ArraysTest > objectCopyOfRangeAllocatesArrayWithOriginalComponentType() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ArraysTest > objectCopyOfRangeAllocatesArrayWithRequestedComponentType() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CollectionsInnerCheckedCollectionTest > addAllCopiesCompatibleElementsThroughTypedEmptyArray() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CollectionsInnerCheckedCollectionTest > addAllRejectsIncompatibleElementsBeforeMutatingBackingCollection() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CollectionsInnerCheckedMapInnerEntrySetViewTest > entrySetCopiesEntriesIntoNonEmptyTypedArray() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CollectionsInnerCheckedMapTest > putAllCopiesCompatibleEntriesIntoCheckedMap() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CollectionsTest > checkedMapEntrySetToArrayReturnsCheckedEntryViews() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ConcurrentHashMapTest > serializationRoundTripPreservesEntriesAndConcurrentMapBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ConcurrentLinkedQueueTest > serializationRoundTripPreservesFifoOrderAndQueueOperations() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ConcurrentSkipListMapTest > serializationRoundTripPreservesSortedEntriesAndConcurrentMapBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ConcurrentSkipListSetTest > rangeViewsAreBackedAndRespectInclusiveBounds() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ConcurrentSkipListSetTest > cloneCreatesIndependentSetWithSameSortedContents() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ConcurrentSkipListSetTest > descendingViewAndPollOperationsMirrorSortedSetSemantics() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CopyOnWriteArrayListInnerCOWSubListTest > toArrayWithUndersizedTypedArrayAllocatesArrayWithRequestedComponentType() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CopyOnWriteArrayListTest > serializationRoundTripPreservesElementsAndCopyOnWriteBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CopyOnWriteArrayListTest > objectArrayConstructorCopiesInputAndPreservesListOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.DelayQueueTest > drainToAndIteratorRemovalHandleExpiredAndUnexpiredElements() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.DelayQueueTest > pollReturnsOnlyExpiredElementsInDelayOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.LinkedBlockingDequeTest > serializationRoundTripPreservesCapacityOrderAndDequeBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.LinkedBlockingDequeTest > typedArrayConversionAllocatesArrayWithComponentTypeAndDequeOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.LinkedBlockingQueueTest > serializationRoundTripPreservesCapacityOrderAndBlockingQueueBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.LinkedBlockingQueueTest > typedArrayConversionAllocatesArrayWithComponentTypeAndQueueOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.LinkedListTest > typedArrayConversionAllocatesArrayWithComponentTypeAndListOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.LinkedListTest > serializationRoundTripPreservesElementsAndDequeBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.PriorityBlockingQueueTest > comparatorConstructorAndDrainToRetainCustomPriorityOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.PriorityBlockingQueueTest > constructorAndQueueOperationsUsePriorityOrdering() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.PriorityBlockingQueueTest > typedArrayConversionPreservesRuntimeComponentType() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.PriorityQueueTest > objectArrayConversionAllocatesObjectArrayAndSerializationPreservesPriorityOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.TreeMapTest > serializationRoundTripPreservesComparatorKeysAndValues() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.TreeSetTest > serializationRoundTripPreservesComparatorAndElements() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.UtilsTest > collectionToTypedArrayAllocatesComponentTypeWhenSuppliedArrayIsTooSmall() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.UtilsTest > collectionToArrayTrimsWhenReportedSizeIsLargerThanIteration() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.UtilsTest > nanoTimeUsesConfiguredProvider() FAILED


Failures (1):
  JUnit Jupiter:UtilsTest:nanoTimeUsesConfiguredProvider()
    MethodSource [className = 'backport_util_concurrent.backport_util_concurrent.UtilsTest', methodName = 'nanoTimeUsesConfiguredProvider', methodParameterTypes = '']
    => org.opentest4j.AssertionFailedError: 
expected: 37L
 but was: 0L
       java.base@25.0.3/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
       backport_util_concurrent.backport_util_concurrent.UtilsTest.nanoTimeUsesConfiguredProvider(UtilsTest.java:36)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.3/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
       [...]

Test run finished after 19 ms
[        25 containers found      ]
[         0 containers skipped    ]
[        25 containers started    ]
[         0 containers aborted    ]
[        25 containers successful ]
[         0 containers failed     ]
[        39 tests found           ]
[         0 tests skipped         ]
[        39 tests started         ]
[         0 tests aborted         ]
[        38 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/native/nativeTestCompile/backport-util-concurrent.backport-util-concurrent_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 24s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for backport-util-concurrent:backport-util-concurrent:3.1 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 26s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.1 at 2026-05-12T10:06:06Z with exit code 1 after 87s
❌ ERROR: 'native-image run' for 3.1 failed with exit code 1.
FAILED[native-image run][3.1][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.1" ./gradlew clean test -Pcoordinates="backport-util-concurrent:backport-util-concurrent:3.1"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.1" ./gradlew clean test -Pcoordinates="backport-util-concurrent:backport-util-concurrent:3.1"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25727047580/job/75542580995
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=3
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'backport-util-concurrent.backport-util-concurrent_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (6.4s @ 0.18GiB)
 Builder configuration:
 - Java version: 25.0.2+10-LTS, vendor version: Oracle GraalVM 25.1.0-dev+10.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (30.8s @ 0.83GiB)
    8,402 types,   9,411 fields, and  37,501 methods found reachable
      379 types,     230 fields, and   1,709 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
      153 resource accesses registered with 262.39KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (3.8s @ 0.96GiB)
[4/8] Parsing methods...      [**]                                                                      (3.2s @ 1.00GiB)
[5/8] Inlining methods...     [***]                                                                     (2.4s @ 0.63GiB)
[6/8] Compiling methods...    [****]                                                                   (17.5s @ 0.73GiB)
[7/8] Laying out methods...   [**]                                                                      (3.6s @ 0.78GiB)
[8/8] Creating image...       [**]                                                                      (3.3s @ 0.92GiB)
   8.73MiB (33.86%) for code area:    23,770 compilation units
  13.31MiB (51.63%) for image heap:  216,334 objects and 21 resources
   3.74MiB (14.51%) for other data
  25.78MiB in total image size, 23.75MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
 979.21KiB o.g.n.runtime/com.oracle.svm.core.code              2.69MiB byte[] for code metadata
 731.39KiB java.base/java.util                                 2.53MiB byte[] for string data
 408.16KiB java.base/jdk.internal.classfile.impl               1.36MiB java.base/java.lang.String
 329.86KiB java.base/java.lang                                 1.00MiB java.base/java.lang.Class
 243.03KiB edu.emory.mathcs.backport.java.util.concurrent    918.97KiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 236.73KiB java.base/java.util.concurrent                    576.10KiB byte[] for general heap data
 228.84KiB java.base/java.text                               471.28KiB heap alignment
 224.55KiB java.base/com.sun.crypto.provider                 425.81KiB byte[] for reflection metadata
 223.43KiB org.assertj.core.internal.bytebuddy.jar.asm       325.78KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 221.00KiB java.base/java.io                                 298.66KiB java.base/java.util.HashMap$Node
   4.99MiB for 316 more packages                               2.79MiB for 2020 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 18 component(s) is embedded in binary (1.01KiB). 27 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       5.4s (7.3% of total time) in 699 GCs | Peak RSS: 1.60GiB | CPU load: 3.59
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/native/nativeTestCompile/backport-util-concurrent.backport-util-concurrent_tests-tests (executable, 23.75MiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /tmp/native-image-build-output.json (build_info, 1.06KiB)
========================================================================================================================
Finished generating 'backport-util-concurrent.backport-util-concurrent_tests-tests' in 1m 13s.

The build process encountered 4 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

backport_util_concurrent.backport_util_concurrent.AbstractExecutorServiceTest > bulkExecutionUsesAbstractExecutorServiceTaskManagement() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.AbstractExecutorServiceTest > shutdownRejectsNewTasks() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.AbstractExecutorServiceTest > submitWrapsRunnableAndCallableTasksInFutures() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ArrayBlockingQueueTest > typedArrayConversionAllocatesArrayWithComponentTypeAndQueueOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ArrayDequeTest > serializationRoundTripPreservesOrderAndDequeBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ArrayDequeTest > typedArrayConversionAllocatesArrayWithComponentTypeAndDequeOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ArraysTest > objectCopyOfRangeAllocatesArrayWithOriginalComponentType() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ArraysTest > objectCopyOfRangeAllocatesArrayWithRequestedComponentType() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CollectionsInnerCheckedCollectionTest > addAllCopiesCompatibleElementsThroughTypedEmptyArray() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CollectionsInnerCheckedCollectionTest > addAllRejectsIncompatibleElementsBeforeMutatingBackingCollection() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CollectionsInnerCheckedMapInnerEntrySetViewTest > entrySetCopiesEntriesIntoNonEmptyTypedArray() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CollectionsInnerCheckedMapTest > putAllCopiesCompatibleEntriesIntoCheckedMap() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CollectionsTest > checkedMapEntrySetToArrayReturnsCheckedEntryViews() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ConcurrentHashMapTest > serializationRoundTripPreservesEntriesAndConcurrentMapBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ConcurrentLinkedQueueTest > serializationRoundTripPreservesFifoOrderAndQueueOperations() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ConcurrentSkipListMapTest > serializationRoundTripPreservesSortedEntriesAndConcurrentMapBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ConcurrentSkipListSetTest > rangeViewsAreBackedAndRespectInclusiveBounds() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ConcurrentSkipListSetTest > cloneCreatesIndependentSetWithSameSortedContents() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ConcurrentSkipListSetTest > descendingViewAndPollOperationsMirrorSortedSetSemantics() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CopyOnWriteArrayListInnerCOWSubListTest > toArrayWithUndersizedTypedArrayAllocatesArrayWithRequestedComponentType() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CopyOnWriteArrayListTest > serializationRoundTripPreservesElementsAndCopyOnWriteBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CopyOnWriteArrayListTest > objectArrayConstructorCopiesInputAndPreservesListOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.DelayQueueTest > drainToAndIteratorRemovalHandleExpiredAndUnexpiredElements() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.DelayQueueTest > pollReturnsOnlyExpiredElementsInDelayOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.LinkedBlockingDequeTest > serializationRoundTripPreservesCapacityOrderAndDequeBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.LinkedBlockingDequeTest > typedArrayConversionAllocatesArrayWithComponentTypeAndDequeOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.LinkedBlockingQueueTest > serializationRoundTripPreservesCapacityOrderAndBlockingQueueBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.LinkedBlockingQueueTest > typedArrayConversionAllocatesArrayWithComponentTypeAndQueueOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.LinkedListTest > typedArrayConversionAllocatesArrayWithComponentTypeAndListOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.LinkedListTest > serializationRoundTripPreservesElementsAndDequeBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.PriorityBlockingQueueTest > comparatorConstructorAndDrainToRetainCustomPriorityOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.PriorityBlockingQueueTest > constructorAndQueueOperationsUsePriorityOrdering() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.PriorityBlockingQueueTest > typedArrayConversionPreservesRuntimeComponentType() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.PriorityQueueTest > objectArrayConversionAllocatesObjectArrayAndSerializationPreservesPriorityOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.TreeMapTest > serializationRoundTripPreservesComparatorKeysAndValues() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.TreeSetTest > serializationRoundTripPreservesComparatorAndElements() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.UtilsTest > collectionToTypedArrayAllocatesComponentTypeWhenSuppliedArrayIsTooSmall() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.UtilsTest > collectionToArrayTrimsWhenReportedSizeIsLargerThanIteration() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.UtilsTest > nanoTimeUsesConfiguredProvider() FAILED


Failures (1):
  JUnit Jupiter:UtilsTest:nanoTimeUsesConfiguredProvider()
    MethodSource [className = 'backport_util_concurrent.backport_util_concurrent.UtilsTest', methodName = 'nanoTimeUsesConfiguredProvider', methodParameterTypes = '']
    => org.opentest4j.AssertionFailedError: 
expected: 37L
 but was: 0L
       java.base@25.0.2/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
       backport_util_concurrent.backport_util_concurrent.UtilsTest.nanoTimeUsesConfiguredProvider(UtilsTest.java:36)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
       [...]

Test run finished after 11 ms
[        25 containers found      ]
[         0 containers skipped    ]
[        25 containers started    ]
[         0 containers aborted    ]
[        25 containers successful ]
[         0 containers failed     ]
[        39 tests found           ]
[         0 tests skipped         ]
[        39 tests started         ]
[         0 tests aborted         ]
[        38 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/native/nativeTestCompile/backport-util-concurrent.backport-util-concurrent_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 17s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for backport-util-concurrent:backport-util-concurrent:3.1 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 19s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.1 at 2026-05-12T10:05:07Z with exit code 1 after 79s
❌ ERROR: 'native-image run' for 3.1 failed with exit code 1.
FAILED[native-image run][3.1][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.1" ./gradlew clean test -Pcoordinates="backport-util-concurrent:backport-util-concurrent:3.1"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.1" ./gradlew clean test -Pcoordinates="backport-util-concurrent:backport-util-concurrent:3.1"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25727047580/job/75542580984
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
n/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=3
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'backport-util-concurrent.backport-util-concurrent_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (6.2s @ 0.18GiB)
 Builder configuration:
 - Java version: 25.0.2+10-LTS, vendor version: Oracle GraalVM 25.1.0-dev+10.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [******]                                                                 (35.0s @ 0.93GiB)
    8,946 types,  10,214 fields, and  40,544 methods found reachable
      507 types,     230 fields, and   1,921 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
      155 resource accesses registered with 262.53KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (4.3s @ 1.06GiB)
[4/8] Parsing methods...      [**]                                                                      (4.4s @ 0.76GiB)
[5/8] Inlining methods...     [***]                                                                     (1.9s @ 0.80GiB)
[6/8] Compiling methods...    [****]                                                                   (20.7s @ 0.82GiB)
[7/8] Laying out methods...   [**]                                                                      (3.0s @ 1.02GiB)
[8/8] Creating image...       [**]                                                                      (3.5s @ 1.15GiB)
   9.86MiB (33.97%) for code area:    26,092 compilation units
  15.13MiB (52.12%) for image heap:  256,105 objects and 23 resources
   4.04MiB (13.91%) for other data
  29.02MiB in total image size, 26.81MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.08MiB o.g.n.runtime/com.oracle.svm.core.code              3.04MiB byte[] for code metadata
 737.69KiB java.base/java.util                                 2.83MiB byte[] for string data
 408.13KiB java.base/jdk.internal.classfile.impl               1.50MiB java.base/java.lang.String
 339.32KiB java.base/java.lang                                 1.07MiB java.base/java.lang.Class
 267.01KiB java.base/com.sun.crypto.provider                 978.47KiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 243.21KiB edu.emory.mathcs.backport.java.util.concurrent    924.69KiB byte[] for general heap data
 236.84KiB java.base/java.util.concurrent                    494.13KiB byte[] for reflection metadata
 235.30KiB java.base/java.io                                 366.73KiB java.base/java.lang.Object[]
 231.40KiB java.base/java.text                               347.03KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 223.43KiB org.assertj.core.internal.bytebuddy.jar.asm       314.41KiB java.base/j.u.concurrent.ConcurrentHashMap$Node
   5.92MiB for 327 more packages                               3.33MiB for 2179 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 18 component(s) is embedded in binary (1.01KiB). 27 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       6.3s (7.7% of total time) in 845 GCs | Peak RSS: 1.80GiB | CPU load: 3.63
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/native/nativeTestCompile/backport-util-concurrent.backport-util-concurrent_tests-tests (executable, 26.81MiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /tmp/native-image-build-output.json (build_info, 1.06KiB)
========================================================================================================================
Finished generating 'backport-util-concurrent.backport-util-concurrent_tests-tests' in 1m 21s.

The build process encountered 4 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

backport_util_concurrent.backport_util_concurrent.AbstractExecutorServiceTest > bulkExecutionUsesAbstractExecutorServiceTaskManagement() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.AbstractExecutorServiceTest > shutdownRejectsNewTasks() SUCCESSFUL


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/native/nativeTestCompile/backport-util-concurrent.backport-util-concurrent_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 24s
backport_util_concurrent.backport_util_concurrent.AbstractExecutorServiceTest > submitWrapsRunnableAndCallableTasksInFutures() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ArrayBlockingQueueTest > typedArrayConversionAllocatesArrayWithComponentTypeAndQueueOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ArrayDequeTest > serializationRoundTripPreservesOrderAndDequeBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ArrayDequeTest > typedArrayConversionAllocatesArrayWithComponentTypeAndDequeOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ArraysTest > objectCopyOfRangeAllocatesArrayWithOriginalComponentType() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ArraysTest > objectCopyOfRangeAllocatesArrayWithRequestedComponentType() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CollectionsInnerCheckedCollectionTest > addAllCopiesCompatibleElementsThroughTypedEmptyArray() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CollectionsInnerCheckedCollectionTest > addAllRejectsIncompatibleElementsBeforeMutatingBackingCollection() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CollectionsInnerCheckedMapInnerEntrySetViewTest > entrySetCopiesEntriesIntoNonEmptyTypedArray() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CollectionsInnerCheckedMapTest > putAllCopiesCompatibleEntriesIntoCheckedMap() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CollectionsTest > checkedMapEntrySetToArrayReturnsCheckedEntryViews() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ConcurrentHashMapTest > serializationRoundTripPreservesEntriesAndConcurrentMapBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ConcurrentLinkedQueueTest > serializationRoundTripPreservesFifoOrderAndQueueOperations() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ConcurrentSkipListMapTest > serializationRoundTripPreservesSortedEntriesAndConcurrentMapBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ConcurrentSkipListSetTest > rangeViewsAreBackedAndRespectInclusiveBounds() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ConcurrentSkipListSetTest > cloneCreatesIndependentSetWithSameSortedContents() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.ConcurrentSkipListSetTest > descendingViewAndPollOperationsMirrorSortedSetSemantics() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CopyOnWriteArrayListInnerCOWSubListTest > toArrayWithUndersizedTypedArrayAllocatesArrayWithRequestedComponentType() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CopyOnWriteArrayListTest > serializationRoundTripPreservesElementsAndCopyOnWriteBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.CopyOnWriteArrayListTest > objectArrayConstructorCopiesInputAndPreservesListOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.DelayQueueTest > drainToAndIteratorRemovalHandleExpiredAndUnexpiredElements() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.DelayQueueTest > pollReturnsOnlyExpiredElementsInDelayOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.LinkedBlockingDequeTest > serializationRoundTripPreservesCapacityOrderAndDequeBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.LinkedBlockingDequeTest > typedArrayConversionAllocatesArrayWithComponentTypeAndDequeOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.LinkedBlockingQueueTest > serializationRoundTripPreservesCapacityOrderAndBlockingQueueBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.LinkedBlockingQueueTest > typedArrayConversionAllocatesArrayWithComponentTypeAndQueueOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.LinkedListTest > typedArrayConversionAllocatesArrayWithComponentTypeAndListOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.LinkedListTest > serializationRoundTripPreservesElementsAndDequeBehavior() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.PriorityBlockingQueueTest > comparatorConstructorAndDrainToRetainCustomPriorityOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.PriorityBlockingQueueTest > constructorAndQueueOperationsUsePriorityOrdering() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.PriorityBlockingQueueTest > typedArrayConversionPreservesRuntimeComponentType() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.PriorityQueueTest > objectArrayConversionAllocatesObjectArrayAndSerializationPreservesPriorityOrder() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.TreeMapTest > serializationRoundTripPreservesComparatorKeysAndValues() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.TreeSetTest > serializationRoundTripPreservesComparatorAndElements() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.UtilsTest > collectionToTypedArrayAllocatesComponentTypeWhenSuppliedArrayIsTooSmall() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.UtilsTest > collectionToArrayTrimsWhenReportedSizeIsLargerThanIteration() SUCCESSFUL

backport_util_concurrent.backport_util_concurrent.UtilsTest > nanoTimeUsesConfiguredProvider() FAILED


Failures (1):
  JUnit Jupiter:UtilsTest:nanoTimeUsesConfiguredProvider()
    MethodSource [className = 'backport_util_concurrent.backport_util_concurrent.UtilsTest', methodName = 'nanoTimeUsesConfiguredProvider', methodParameterTypes = '']
    => org.opentest4j.AssertionFailedError: 
expected: 37L
 but was: 0L
       java.base@25.0.2/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
       backport_util_concurrent.backport_util_concurrent.UtilsTest.nanoTimeUsesConfiguredProvider(UtilsTest.java:36)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
       [...]

Test run finished after 11 ms
[        25 containers found      ]
[         0 containers skipped    ]
[        25 containers started    ]
[         0 containers aborted    ]
[        25 containers successful ]
[         0 containers failed     ]
[        39 tests found           ]
[         0 tests skipped         ]
[        39 tests started         ]
[         0 tests aborted         ]
[        38 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for backport-util-concurrent:backport-util-concurrent:3.1 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 26s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.1 at 2026-05-12T10:06:03Z with exit code 1 after 87s
❌ ERROR: 'native-image run' for 3.1 failed with exit code 1.
FAILED[native-image run][3.1][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.1" ./gradlew clean test -Pcoordinates="backport-util-concurrent:backport-util-concurrent:3.1"]
```

--- a/stats/backport-util-concurrent/backport-util-concurrent/3.1/stats.json
+++ b/stats/backport-util-concurrent/backport-util-concurrent/3.1/stats.json
@@ -15,21 +15,21 @@
     },
     "libraryCoverage" : {
       "instruction" : {
-        "covered" : 7859,
-        "missed" : 29581,
-        "ratio" : 0.209909,
+        "covered" : 7812,
+        "missed" : 29628,
+        "ratio" : 0.208654,
         "total" : 37440
       },
       "line" : {
-        "covered" : 1961,
-        "missed" : 6662,
-        "ratio" : 0.227415,
+        "covered" : 1960,
+        "missed" : 6663,
+        "ratio" : 0.227299,
         "total" : 8623
       },
       "method" : {
-        "covered" : 461,
-        "missed" : 1763,
-        "ratio" : 0.207284,
+        "covered" : 460,
+        "missed" : 1764,
+        "ratio" : 0.206835,
         "total" : 2224
       }
     }

--- a/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build.gradle
+++ b/tests/src/backport-util-concurrent/backport-util-concurrent/3.1/build.gradle
@@ -16,17 +16,16 @@ tasks.withType(Test).configureEach {
     systemProperty nanoTimerProviderProperty, nanoTimerProviderClass
 }
 
+tasks.named("nativeTest") {
+    runtimeArgs.add("-D${nanoTimerProviderProperty}=${nanoTimerProviderClass}")
+}
+
 dependencies {
     testImplementation "backport-util-concurrent:backport-util-concurrent:$libraryVersion"
     testImplementation 'org.assertj:assertj-core:3.22.0'
 }
 
 graalvmNative {
-    binaries {
-        test {
-            runtimeArgs("-D${nanoTimerProviderProperty}=${nanoTimerProviderClass}")
-        }
-    }
     agent {
         defaultMode = "conditional"
         modes {

